File: app/src/main/java/ru/rinet/questik/data/room/dao/MetricsDao.java
package ru.rinet.questik.data.room.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ru.rinet.questik.data.room.models.Corp;
import ru.rinet.questik.data.room.models.Metrics;

@Dao
public interface MetricsDao {
    @Query("SELECT * FROM Metrics")
    List<Metrics> getAll();

    //запрос на каегорию по его id
    @Query("SELECT * FROM Metrics WHERE metrics_id = :id")
    Metrics getById(Long id);

    @Query("SELECT * FROM Metrics")
    Metrics getMetrics();

    @Query("SELECT * FROM Metrics")
    List<Metrics> getMetricsList();

    //Wipe
    @Query("DELETE FROM Metrics")
    void wipeMetricsTable();

    @Insert
    Long insert(Metrics metrics);

    @Insert
    List<Long> insertMetricsList(List<Metrics> metricsList);

    @Update
    int update(Metrics metrics);

    @Delete
    int delete(Metrics metrics);

}

File: app/src/main/java/xyz/gmfatoom/questik/repo/local/room/dao/MetricsDao.kt
package xyz.gmfatoom.questik.repo.local.room.dao

import androidx.paging.PagingSource
import androidx.room.*
import kotlinx.coroutines.flow.Flow
import xyz.gmfatoom.questik.repo.local.room.entity.MetricsEntity

@Dao
interface MetricsDao {

    @Query("SELECT COUNT(id) FROM metrics")
    fun getMetricsCount(): Int
    @Query("SELECT COUNT(id) FROM metrics")
    fun getMetricsCountFlow(): Flow<Int>

    @Query("SELECT * FROM metrics")
    fun getAllMetrics(): PagingSource<Int, MetricsEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllMetrics(items:List<MetricsEntity>)

    @Query("DELETE FROM metrics")
    fun clearMetrics()

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertMetrics(metrics: MetricsEntity)

    @Update
    fun updateMetrics(metrics: MetricsEntity)

    @Delete
    fun deleteMetrics(metrics: MetricsEntity)

    @Query("SELECT * FROM metrics WHERE id == :id")
    fun getMetricsById(id: Int): MetricsEntity

    @Query("SELECT * FROM metrics")
    fun getMetrics(): List<MetricsEntity>
}